bundle-workflow/src/test_workflow/test_recorder.py

```diff
@@ -3,34 +3,142 @@
-        FAILED means the tests ran and at least one failed. ERROR means the test suite did not run correctly.
+        :param component_test_config: component_config under consideration for test eg: with/without-security.
+        :param exit_code: Integer value of the exit code.
         :param stdout: A string containing the stdout stream from the test process.
         :param stderr: A string containing the stderr stream from the test process.
         :param results: A generator that yields tuples containing test results files, in the form (absolute_path, relative_path).
         """
+
+        base = self.__create_base_folder_structure(component_name, component_test_config)
+        dest_directory = os.path.join(base, "test_outcome")
+        os.makedirs(dest_directory, exist_ok=False)
         logging.info(
-            f"Recording test results for {component_name}. Exit status: {exit_status}, stdout: {stdout}, stderr: {stderr}, results files: {results}"
-        )
+            f"Recording component test results for {component_name} at {os.path.realpath(dest_directory)}")
+        self.__generate_std_files(stdout, stderr, dest_directory)
+        results_dir = list(results)
+        for result in results_dir:
+            dest_file = os.path.join(dest_directory, os.path.basename(result[0]))
+            shutil.copyfile(result[0], dest_file)
+        exit_status = self.__get_exit_status(exit_code)
+        component_yml = self.__generate_test_outcome_yml(component_name, component_test_config, exit_status, "S3")
+        shutil.copyfile(component_yml, os.path.join(dest_directory, os.path.basename(component_yml)))
+
+    def __create_base_folder_structure(self, component_name, component_test_config):
+        dest_directory = os.path.join(self.location, "tests", self.test_run_id, self.test_type, str(component_name),
+                                      str(component_test_config))
+        os.makedirs(dest_directory, exist_ok=True)
+        return os.path.realpath(dest_directory)
+
+    def __get_exit_status(self, exit_code):
+        if exit_code == 0:
+            return "SUCCESS"
+        else:
+            return "FAILED/ERROR"
+
+    def __generate_test_outcome_yml(self, component_name, component_test_config, exit_status, log_file_location):
+        dict_file = {
+            "test_type": self.test_type,
+            "test_run_id": self.test_run_id,
+            "component_name": component_name,
+            "test_config": component_test_config,
+            "status": exit_status,
+            "log_file_location": log_file_location
+        }
+        with open("%s.yml" % component_name, "w") as file:
+            yaml.dump(dict_file, file)
+        return os.path.realpath("%s.yml" % component_name)
+
+    def __generate_std_files(self, stdout, stderr, location):
+        stdout_file = open(os.path.join(location, "stdout.txt"), "w")
```

Comment:
Use `with`, so you won't need a try/finally

Reply:
Done